glide.yaml

```diff
@@ -223,16 +253,27 @@ import:
   - test/e2e/chaosmonkey
   - test/e2e/common
   - test/e2e/framework
+  - test/e2e/framework/ginkgowrapper
+  - test/e2e/framework/metrics
+  - test/e2e/manifest
   - test/e2e_node/services
   - test/integration/framework
-  - test/utils/image
+  - test/utils
   - third_party/forked/etcd221/wal
 
 # The versions of the following dependencies need to correspond to the
 # versions used by the pinned version of k8s.io/kubernetes
 - package: github.com/jteeuwen/go-bindata
   version: a0ff2567cfb70903282db057e799fd826784d41d
+- package: k8s.io/apiextensions-apiserver
+  version: kubernetes-1.9.2-beta.0
 - package: k8s.io/code-generator
-  version: 961bc077103364eb5bda2c40e2b560d068c9a8c6
+  version: kubernetes-1.9.2-beta.0
 - package: k8s.io/gengo
-  version: 70ad626ed2d7a483d89d2c4c56364d60b48ee8fc
+  version: b58fc7edb82e0c6ffc9b8aef61813c7261b785d4
+- package: k8s.io/kube-aggregator
+  version: kubernetes-1.9.2-beta.0
+- package: k8s.io/metrics
+  version: kubernetes-1.9.2-beta.0
+- package: k8s.io/utils
+  version: aedf551cdb8b0119df3a19c65fde413a13b34997
```

Comment:
Is this autofilled by glide? I see that some k8s.io repos still don't have tags mapping to a particular k8s release tag (perhaps not required). But I do not clearly understand how the compatibility mapping is maintained for these as of now. Does it make sense to have any discussion on this?

Reply:
The version is not auto-filled by glide. It is manually added by looking into Godeps.Json from kubernetes repo for the specific version we are bumping. Most of the repos under kubernetes are tagged now for the release. Going forward we can write a simple tool to pick up the versions and populate the glide.yaml file.

Reply:
Out of curiosity, why are you looking to manually control the version of dependencies from k8s rather than let glide pick them up from the godeps.json file? Glide walks the tree and when it finds a dependency it uses the information picked up along the way on dependency versions. If k8s lists a dependency revision in the godeps.json file before others list it in their metadata files it will use that version.

Imagine the case where you manually manage the version and k8s changes. Then you'll be out of sync with k8s on the version to use.

There are plenty of reasons to manually manage a dependency. I'm curious of the reason here.

Reply:
@mattfarina, federation also imports some third party packages which might be imported by k8s version and if glide resorts to using latest version of the code for such packages. it might cause some incompatibility. ( I should admit i don't know how glide handles this scenario). To ensure that we are vendoring the exact same version of the package which kubernetes version uses, i resorted to explicitly specifying version in the package. kubernetes code makes up of the bulk of vendored code what we use. and we always rely on a particular kubernetes version (which is stable enough).
IMHO, In our case, Its just a bit of overhead to change all the version of the package which we vendor when we bump.
I would also do some experiments in glide and confirm, if we can do away not mentioning version for every package.

Reply:
If you have kubernetes first in your glide.yaml than the dependency versions listed in the Kubernetes Godeps.json file will be used if no version is specified in the glide.yaml file. This is because Glide will start at the first one and iterate over the dependencies first to last in the glide.yaml file. When it picks up a version the first one wins (unless there are some semver ranges it can munge together which isn't the case with Godeps).

The ordering in Glide was an intentional decision to give developers some control.

Reply:
Thanks for this helpful tip @mattfarina. i will update accordingly.